### PR TITLE
Handle MTU migration

### PR DESCRIPTION
**Pull request by danwinship**
Implements https://issues.redhat.com/browse/SDN-2553
Depends on https://github.com/openshift/cluster-network-operator/pull/1241

/cc @Kyrtapz @jcaamano

### CODE REVIEW

**Comment by Kyrtapz** [comment]
There is an issue with setting up the routes:
`error adding container to network "openshift-sdn": failed to add route to 10.128.0.0/14 via SDN: no such device`

I think that it is related to the fact that there is no link specified [here](https://github.com/openshift/sdn/pull/381/files#diff-8b4e24ba0764957c1addd72fceac5346db44409597863cd4884aa95ea885decdR240-R247).
`ConfigureIface` always specifies a link when adding routes:
```
func ConfigureIface(ifName string, res *current.Result) error {
...
		if err = ip.AddRoute(&r.Dst, gw, link); err != nil {
			// we skip over duplicate routes as we assume the first one wins
			if !os.IsExist(err) {
				return fmt.Errorf("failed to add route '%v via %v dev %v': %v", r.Dst, gw, ifName, err)
			}
		}
	}
```

**Comment by jcaamano** [comment]
I am missing the routable-mtu being set on the service network route on the host itself.

File: pkg/cmd/openshift-sdn-cni/openshift-sdn.go [hunk 1]
```diff
@@ -233,18 +229,22 @@ func (p *cniPlugin) CmdAdd(args *skel.CmdArgs) error {
...
 			}
 		}
 
-		dsts = append(dsts, serviceIPNet)
-		for _, dst := range dsts {
+		for _, cniroute := range routes {
 			route := &netlink.Route{
-				Dst: dst,
-				Gw:  defaultGW,
+				LinkIndex: link.Attrs().Index,
```
Comment (Kyrtapz) [inline pkg/cmd/openshift-sdn-cni/openshift-sdn.go:241]: I think this should be a link of `args.IfName`, not `macvlan0`

**Comment by jcaamano** [comment]
@danwinship We need to use the migration mtu here as well during migration, otherwise observing br0 and tun0 with an incorrect mtu.
https://github.com/openshift/sdn/blob/99d96f8a65b6ae9c7a33ebf3d7d2391101b213ad/pkg/network/node/sdn_controller.go#L157-L159

**Review by alexanderConstantinescu — APPROVED** [review]
/lgtm

**Pull request merged**